### PR TITLE
closes #5880. Adding doc how to setup azure auth with cli

### LINKED ISSUE #5880: Azure: Documentation on Authentication
The documentation is sparse on how to correctly authenticate the azure provides.

See also [this comment](https://github.com/hashicorp/terraform/issues/5875#issuecomment-202097500) on #5875 - it is also suggested that the issue might result in incorrectly setup authentication.  Which is hard to do with the current sparse information in the documentation.

I am trying this guide: https://msdn.microsoft.com/en-us/library/azure/ee460782.aspx#bk_ad but setting up Azure AD just to access the API seems not to be the "right way" to do it, or is it?

### PR DESCRIPTION
**Pull request by majormoses**

### CODE REVIEW

File: website/source/docs/providers/azurerm/index.html.markdown [hunk 1]
```diff
@@ -115,7 +115,24 @@ To grant permissions to the App Registration to your subscription, you now must
 - Select **Subscriptions** from the left panel. Select the subscription that you want to use. In the Subscription details pane, click **Access Control (IAM)**
 - Click **Add**.  For Step 1 select an appropriate role for the tasks you want to complete with Terraform. You can find details on the built in roles [here](https://azure.microsoft.com/en-gb/documentation/articles/role-based-access-built-in-roles/)
 - Type in the name of the application added in the search box. You need to type this as it won't be shown in the user list. Click on the appropriate user in the list and then click **Select**
-- Click **OK** in the **Add Access** panel. The changes will now be saved   
+- Click **OK** in the **Add Access** panel. The changes will now be saved
+
+To create using azure cli:
+```
+az login
```
Comment (pmcatominey) [inline website/source/docs/providers/azurerm/index.html.markdown]: Is `az` a custom alias, I only have `azure` available as a command out of the box.

Comment (majormoses) [inline website/source/docs/providers/azurerm/index.html.markdown]: not sure I based it on the doc I found...I need to validate...I am hoping this is not another command/utility they are using.

Comment (majormoses) [inline website/source/docs/providers/azurerm/index.html.markdown]: I have not used azure in a while I will see what I can come up with.

Comment (stack72) [inline website/source/docs/providers/azurerm/index.html.markdown]: Yeah `az` is definitely an alias from what i can see:

```
stacko@Pauls-MacBook-Pro:~/Code/engineering-services-org/engservices-teamcity on master]
% az
zsh: command not found: az
[stacko@Pauls-MacBook-Pro:~/Code/engineering-services-org/engservices-teamcity on master]
% azure                                                                                                                                                                                               127 ↵
info:             _    _____   _ ___ ___
info:            /_\  |_  / | | | _ \ __|
info:      _ ___/ _ \__/ /| |_| |   / _|___ _ _
info:    (___  /_/ \_\/___|\___/|_|_\___| _____)
info:       (_______ _ _)         _ ______ _)_ _
info:              (______________ _ )   (___ _ _)
```

**Comment by majormoses** [comment]
@stack72 updated to indicate that this uses the new azure cli which is now python based rather than the old nodejs.

**Comment by majormoses** [comment]
https://github.com/hashicorp/terraform/issues/5880#issuecomment-259370480

**Comment by stack72** [comment]
LGTM! Thanks for this :)

**Pull request merged**